### ISSUE TITLE
 Unable to Reproduce Reported Results with OpenCity-Plus Model

### ISSUE THREAD

**Issue by ywy377** [HKUDS/OpenCity#9]
Hello, thank you for the open-source project！
I attempted to reproduce the OpenCity-Plus model as described in the paper, using the specified dataset and hyperparameters. I ran the following pretraining command:
**nohup python Run.py -mode pretrain -model OpenCity -save_pretrain_path OpenCity-plus2.0.pth -batch_size 2 --embed_dim 512 --skip_dim 512 --enc_depth 6 > output.log 2>&1 &**
All parameters are set to the default values as per the code, except for batch_size, which I reduced to 2 to fit within the memory constraints of my GPU (model: NVIDIA GeForce RTX 4090 D). After training for 10 epochs, I evaluated the model but found that the results differ significantly from those reported in the paper.
The following images are provided for reference: my PyTorch version details, the final screenshot of the pretraining, and the evaluation results.
![9U`$42C O8~VH`AW P )DYT](https://github.com/user-attachments/assets/4dac47b1-9440-4621-a13c-bd17947e0fca)
![HFDPHWPK(2DYZQME0H9)2K0](https://github.com/user-attachments/assets/ee5531ee-71fc-44f3-bc3c-eb4563cc5b37)
![%2)TYRBZ A(%}%~L}EIT5SE](https://github.com/user-attachments/assets/92f422e0-4af4-4568-8922-5eeaaeebfed4)

Could you please help me identify what might be causing this discrepancy?
Thank you!

**Comment by LZH-YS1998** [comment]
Hello, thank you for your attention. 
This might be related to the scope of the test set. Could you please confirm whether your test set scope matches what was described in the paper? Perhaps you could try using our provided OpenCity plus model directly to see if you can obtain results similar to those reported in Table 1.

**Comment by ywy377** [comment]
Thank you very much for your reply. I was able to obtain the same test results as in Table 1 using the provided OpenCity plus model. However, I also wanted to try reproducing the model weights myself. Here are the specific steps I took: I only removed the first line's dataset in the pretrain.conf file and commented out the second line, which I confirmed matches the pretraining datasets listed in Table 5. Then, I specified a GPU (model: NVIDIA GeForce RTX 4090 D) by using export **CUDA_VISIBLE_DEVICES=0** and ran the command **nohup python Run.py -mode pretrain -model OpenCity -save_pretrain_path OpenCity-plus2.0.pth -batch_size 2 --embed_dim 512 --skip_dim 512 --enc_depth 6 > output.log 2>&1 &**; I trained for 10 epochs, with each epoch having 171,800 steps, totaling 82.5 hours. 
![~O{46I5U%F0 B5677PK$TBX](https://github.com/user-attachments/assets/f098f6a5-183c-4d0e-92c8-fd026ab7276c)
![$V}_MH8@GB1ATD 2T1`0M99](https://github.com/user-attachments/assets/a90b0a68-dc61-4139-8556-35ab85c5a4d0)

Could you please let me know if there is any issue with my pretraining operation or if I missed any critical steps that might be causing the discrepancy in evaluation results? Looking forward to your response, thank you.

**Comment by LZH-YS1998** [comment]
Thank you for providing the detailed information. From your description of the experimental setup, both the training and test set configurations are correct, with the main difference being the batch_size setting. When training OpenCity plus, we set the batch_size to 24 (8*A100 40G).
﻿
To verify this issue, we conducted preliminary tests with batch_size=8, and indeed observed a performance degradation that falls between your reported batch_size=2 and the original weights' batch_size=24. We believe that batch size may have a significant impact on model performance. We will conduct further experiments with different batch size settings (such as batch_size=8, batch_size=16...) and add the results in the paper, though this may take some time.
﻿
If you want to reproduce OpenCity plus's performance, I suggest using a multi-GPU server with a larger batch size, such as batch_size=24 (the provided code supports multi-GPU mode and can be run directly with the command). Alternatively, you can experiment with smaller-scale versions. Thank you for this discovery.

**Comment by ywy377** [comment]
Thank you very much for your response. I will try experimenting with a multi-GPU server and a larger batch size. By the way, this is my first reproduction experiment, and I’m really happy to receive such a thoughtful reply. Wishing you success in your research.
